File: tests/fax/pjsip/t38_fast_reject/check_reinvite_rtt.pl
#!/usr/bin/perl
use strict;
use Text::CSV;
use Data::Dumper;

my $filename = $ARGV[0];
open(my $fh, '<:utf8', $filename)
    or die "Can't open $filename: $!";

my $csv = Text::CSV->new({sep_char => ';'})
    or die "Text::CSV error: " . Text::CSV->error_diag;

my $header = <$fh>; 
# define column names    
$csv->parse($header);
$csv->column_names([$csv->fields]);

my $ret = 0;
# parse the rest
while (my $row = $csv->getline_hr($fh)) {
    if (!defined  $row->{'ResponseTimereinvite(P)'} ) {
	    print "column not found! make sure scenario is correct!\n";
	    $ret = -1;
	    last;
    }
 
    my $pkey = $row->{'ResponseTimereinvite(P)'};
    my ($hour, $minute, $second, $msec) = split(/:/, $pkey);
    $minute += $hour*60;
    $second += $minute*60;
    $msec += $second * 1000;
    
    if ($msec > 500) {
	    print "Slow 488 Rejection detected ($msec)!\n";
	    $ret = -2;
	    last;
    }
}

$csv->eof or $csv->error_diag;
close $fh;

exit $ret;

File: tests/fax/pjsip/t38_fast_reject/check_reinvite_rtt.py
#!/usr/bin/python

import csv
import sys
import re

regex = re.compile("([0-9]{2}):([0-9]{2}):([0-9]{2}):([0-9]{3})([0-9]{3})?")
with open(sys.argv[1]) as csvfile:
    reader = csv.DictReader(csvfile, delimiter=';')
    for row in reader:
        if not row.has_key('ResponseTimereinvite(P)'):
            print "column not found! make sure scenario is correct!\n"
            exit(-1);
        parts = regex.match(row['ResponseTimereinvite(P)'])
        hours = int(parts.group(1))
        minutes = (hours * 60) + int(parts.group(2))
        seconds = (minutes * 60) + int(parts.group(3))
        milliseconds = (seconds * 1000) + int(parts.group(4))
        if (milliseconds > 500):
            print "Slow 488 Rejection detected (" + milliseconds + ")!\n";
            exit(-2)

csvfile.close()
exit(0)